src/verification/verification.ts

```diff
@@ -77,97 +72,71 @@ export class UbirchVerification {
-              verificationResult.verificationState = EVerificationState.VERIFICATION_SUCCESSFUL;
-
-              this.handleInfo(EInfo.BLXTXS_FOUND_SUCCESS);
-            } else {
-              verificationResult.verificationState =
-                EVerificationState.VERIFICATION_PARTLY_SUCCESSFUL;
-
-              this.handleInfo(EInfo.NO_BLXTX_FOUND);
-            }
-          } catch (err) {
-            verificationResult.verificationState = EVerificationState.VERIFICATION_FAILED;
-            if (err.code) {
-              verificationResult.failReason = err.code;
-            }
-
-            this.handleVerificationState(verificationResult.verificationState, verificationResult);
-            reject(verificationResult);
-          }
+    return this.sendVerificationRequest(hash)
+      .then((verificationResponse: IUbirchVerificationResponse) => {
+        this.handleInfo(EInfo.START_CHECKING_RESPONSE);
+
+        const ubirchUpp: IUbirchUpp = this.extractUpp(verificationResponse);
+
+        verificationResult.upp = ubirchUpp;
+        verificationResult.verificationState = EVerificationState.VERIFICATION_PARTLY_SUCCESSFUL;
+
+        this.handleInfo(EInfo.UPP_HAS_BEEN_FOUND);
+
+        // TODO: check that upp contains given hash
+        // TODO: check signature, ...
+
+        const blxAnchors: IUbirchBlockchainAnchor[] = this.checkBlockchainTXs(verificationResponse);
 
-          this.handleVerificationState(verificationResult.verificationState, verificationResult);
-          resolve(verificationResult);
-        })
-        .catch((err) => {
-          verificationResult.verificationState = EVerificationState.VERIFICATION_FAILED;
+        if (blxAnchors.length > 0) {
+          verificationResult.anchors = blxAnchors;
+          verificationResult.upp.state = EUppStates.anchored;
+          verificationResult.verificationState = EVerificationState.VERIFICATION_SUCCESSFUL;
 
-          verificationResult.failReason = err.code || EError.UNKNOWN_ERROR;
+          this.handleInfo(EInfo.BLXTXS_FOUND_SUCCESS);
+        } else {
+          verificationResult.verificationState = EVerificationState.VERIFICATION_PARTLY_SUCCESSFUL;
 
-          this.handleVerificationState(verificationResult.verificationState, verificationResult);
-          reject(verificationResult);
-        });
-    });
+          this.handleInfo(EInfo.NO_BLXTX_FOUND);
+        }
+
+        this.handleVerificationState(verificationResult.verificationState, verificationResult);
+        return verificationResult;
+      })
+      .catch((err) => {
+        verificationResult.verificationState = EVerificationState.VERIFICATION_FAILED;
+
+        verificationResult.failReason = err.code || EError.UNKNOWN_ERROR;
+
+        this.handleVerificationState(verificationResult.verificationState, verificationResult);
+        return verificationResult;
+      });
   }
 
-  public formatJSON(json: string, sort = true): string {
```

Comment:
we need this sort param, because of backward compatibility - in the first certificates the JSON has not been sorted and the params need to be stringified in the given order of the keys